### DISCUSSION TITLE
[Solved] Axios loses or filters my headers

### DISCUSSION THREAD

**Discussion by Jonathan64340** [axios/axios#5673] [General]
Hello to all of you,
I use SailsJS in backend, 
I have a problem with Axios to get some headers that are visible in my headers but not in my interceptor in the case of a catch.

I try to retrieve the "X-Exit-Description" header that is visible in my catch request in the list of headers but Axios does not return it

Does anyone have an idea?
Thank you very much, Thank you for your help.
![Capture d’écran 2023-04-23 à 19 00 10](https://user-images.githubusercontent.com/25491557/233853773-96325d16-47c7-47f4-9287-7d0b759fa14e.png)
![Capture d’écran 2023-04-23 à 19 01 48](https://user-images.githubusercontent.com/25491557/233853834-6248bcac-b813-4a74-b3d8-54c3ae37f4be.png)
![Capture d’écran 2023-04-23 à 19 03 12](https://user-images.githubusercontent.com/25491557/233853901-3cb6aa68-d8f6-4dae-8583-49d5afdf584d.png)

**Comment by DigitalBrainJS** [comment]
> Does anyone have an idea?

Is CORS enabled?

**Reply by Jonathan64340** [reply]
Yes cors is well configured, in fact I expressed myself badly, I voluntarily inform bad identifiers on my application to be able to recover the header in question "X-Exit-Description" which indicates me that I informed a bad combo of identifiers. This is what I want to retrieve via Axios Error because I fall into the catch of my request as it returns a 401 status. In my networks I can see this header but not in Axios when I log what happens in my interceptor.

**Reply by DigitalBrainJS** [reply]
https://codesandbox.io/p/sandbox/nice-burnell-bsnz6y

**Reply by Jonathan64340** [reply]
it works, however on my sailsjs application I don't have all my headers including this one, which is really strange, yet I see it well on my headers, I don't see what I'm doing wrong.

**Reply by DigitalBrainJS** [reply]
It's unlikely that there could be other reasons than a misconfigured CORS policy. Axios returns whatever the browser passes to it as headers. ¯¯\_(ヅ)_/¯

**Reply by Jonathan64340** [reply]
![Capture d’écran 2023-04-23 à 20 31 09](https://user-images.githubusercontent.com/25491557/233858252-a899fa3c-ab8c-4bcb-b303-bcd52242d9fb.png)
I see only that headers on error.response.headers ...

**Reply by Jonathan64340** [reply]
This is the first time I've had this problem, I'm aware that as soon as the header is sent back, the browser gets it

**Comment by Jonathan64340** [comment]
![Capture d’écran 2023-04-23 à 20 34 43](https://user-images.githubusercontent.com/25491557/233858417-7f2de642-4a45-4c42-8d10-100b9a0a2713.png)
Here is my cors configuration on sailsjs

**Reply by Jonathan64340** [reply]
I have find the solution ! Thank you for your investigation, I just had to tell him that I wanted to exhibit a particular header
![Capture d’écran 2023-04-23 à 20 38 05](https://user-images.githubusercontent.com/25491557/233858549-e2b753fb-d50b-4bd8-9193-1bec4a4dec03.png)

**Reply by DigitalBrainJS** [reply]
Yes, that's it. Almost everything mysteriously missing comes down to CORS 🙂
